### DISCUSSION TITLE
call_next method takes longer than the actual time required

### DISCUSSION THREAD

**Discussion by juyunsang** [fastapi/fastapi#8494] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import APIRouter, Request
from fastapi import FastAPI
from fastapi_utils.cbv import cbv

app = FastAPI()

@app.middleware('http')
async def log_middleware(request: Request, call_next):
    request.state.timer = Timer()
    request.state.timer.start('callNextTime')
    response = await call_next(request)
    request.state.timer.end('callNextTime')

router = APIRouter()
@cbv(router)
class TestRouter:
    @router.post('/')
    async def post(self, request:Request):
        request.state.timer.start('routerTime')
        # Do something
        request.state.timer.end('routerTime')
        return response
```

### Description

When we run the stress test, the callNextTime takes much longer than the routerTime
I'd like to ask why this phenomenon occurs
![image](https://user-images.githubusercontent.com/13113520/169445185-f3bee00f-1cf2-436b-8038-58eee3eb877e.png)

### Operating System

Linux

### FastAPI Version

0.65.2

### Python Version

3.8.5

**Comment by jhonnattan123** [comment]
most likely the method inside the await 'response = await call_next(request)' is performing an operation that takes too long. i invite you to read the async await documentation.

https://fastapi.tiangolo.com/async/

**Comment by jgould22** [comment]
Are you blocking the event loop at any point in your code? (or using a lib that is)

**Comment by rafsaf** [comment]
It's expected that it takes longer obviously, between callNextTime there is **whole** FastAPI magic like machinery of handling requests, when at the same time routerTime is only your piece of code and nothing more.

```
from fastapi import APIRouter, Request
from fastapi import FastAPI

from time import time

app = FastAPI()


@app.middleware("http")
async def log_middleware(request: Request, call_next):
    request.state.callNextTime = time()
    response = await call_next(request)
    request.state.callNextTime = time() - request.state.callNextTime

    print(request.state.routerTime)
    print(request.state.callNextTime)
    return response


router = APIRouter()


@router.post("/")
async def post(request: Request):
    request.state.routerTime = time()
    # Do something
    x = 0
    for i in range(10000):
        x += i
    request.state.routerTime = time() - request.state.routerTime

    return x


app.include_router(router)

```
For above similar snippet i get following results (uvicorn)
0.0008385181427001953
0.001611948013305664

~2x difference, in your example it is ~5-7x, I have no idea what is `from fastapi_utils.cbv import cbv` and what is does, I would look into it first, because almost for sure it is your bottleneck here.

**Comment by Kludex** [comment, marked as answer]
https://github.com/encode/starlette/pull/1640
